src/transactionWatcher.ts

```diff
@@ -152,6 +161,11 @@ export class TransactionWatcher {
             }
         }
 
+        // The patience timer isn't subject to the timeout constraints.
+        if (satisfied) {
+            await patienceTimer.start(this.patienceMilliseconds);
```

Comment:
Just providing some feedback here, shouldn't `doFetch` be called after waiting for `patienceTimer` so `fetchedData` is actually the up to date data (contains smart contract results/logs etc if that was the usecase) ?